### PR TITLE
[Train] Update PyTorch Lightning import path

### PR DESCRIPTION
**Pull request by woshiyyya**
## Why are these changes needed?

PyTorch Lightning 2.0 updated the import path from `pytorch_lightning` to `lightning.pytorch`. This change will first detect if the latest import path exists, then fallback to the original one.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/train/lightning/_lightning_utils.py [hunk 1]
```diff
@@ -20,19 +20,21 @@
...
 from torch.utils.data import IterableDataset, DataLoader
 
-import pytorch_lightning as pl
-from pytorch_lightning.callbacks import ModelCheckpoint, Callback
-from pytorch_lightning.plugins.environments import LightningEnvironment
-from pytorch_lightning.strategies import DDPStrategy, DeepSpeedStrategy
+try:
+    import lightning.pytorch as pl
+except ModuleNotFoundError:
+    import pytorch_lightning as pl
```
Comment (matthewdeng) [inline python/ray/train/lightning/_lightning_utils.py:26]: Can we wrap this as a utility function, e.g.

```python
def import_lightning():
    try:
        import lightning.pytorch as pl
    except ModuleNotFoundError:
        import pytorch_lightning as pl
    return pl

pl = import_lightning()
```

File: python/ray/train/lightning/__init__.py [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
-        "PyTorch Lightning isn't installed. To install PyTorch Lightning, "
-        "please run 'pip install pytorch-lightning'"
-    )
+    try:
+        import pytorch_lightning  # noqa: F401
+    except ModuleNotFoundError:
+        raise ModuleNotFoundError(
+            "PyTorch Lightning isn't installed. To install PyTorch Lightning, "
+            "please run 'pip install lightning'"
+        )
```
Comment (matthewdeng) [inline python/ray/train/lightning/__init__.py:11]: nit: would it make sense to include this in `import_lightning()`?

File: python/ray/train/lightning/_lightning_utils.py [hunk 3]
```diff
@@ -57,7 +62,7 @@ def get_worker_root_device():
 
 
 @PublicAPI(stability="beta")
-class RayDDPStrategy(DDPStrategy):
+class RayDDPStrategy(pl.strategies.DDPStrategy):
```
Comment (matthewdeng) [inline python/ray/train/lightning/_lightning_utils.py:65]: Curious, would it make sense to import this after line 29?

**Review by matthewdeng — APPROVED** [review]
Awesome ⚡

File: python/ray/train/lightning/_lightning_utils.py (hunk 3 above)
Comment (woshiyyya) [reply python/ray/train/lightning/_lightning_utils.py:65]: Although we can use `pl` as the prefix of the class path. We can not do `from pl.strategies import DDPStrategy`

**Comment by woshiyyya** [comment]
Rerun the release tests that are using lightning 2.0

- dolly v2: https://buildkite.com/ray-project/release-tests-pr/builds/53903
- vicuna: TODO

File: python/ray/train/lightning/_lightning_utils.py (hunk 3 above)
Comment (matthewdeng) [reply python/ray/train/lightning/_lightning_utils.py:65]: Oh why?

Comment (woshiyyya) [reply python/ray/train/lightning/_lightning_utils.py:65]: hmm, this is the python syntax. We cannot import from an alias, but the full module path.

**Review by aslonnie — APPROVED** [review]

**Comment by woshiyyya** [comment]
Ok. Both release tests passed!

Resolved the merge conflicts and run the full CI again.

**Review by matthewdeng — APPROVED** [review]

**Pull request merged**